File: src/ros_study_cpp/src/my_msg_subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include "ros_study_msgs/msg/my_msg.hpp"

using std::placeholders::_1;

class MyMsgSubscriber : public rclcpp::Node
{
public:
  MyMsgSubscriber()
  : Node("my_msg_subscriber")
  {
    auto qos = rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(rmw_qos_profile_default)); //10
    qos.keep_last(10); //10

    subscription_ = this->create_subscription<ros_study_msgs::msg::MyMsg>(
      "MyMsg", qos, std::bind(&MyMsgSubscriber::topic_callback, this, _1));
  }

private:
  rclcpp::Subscription<ros_study_msgs::msg::MyMsg>::SharedPtr subscription_;
  void topic_callback(const ros_study_msgs::msg::MyMsg::SharedPtr msg)
  {
    RCLCPP_INFO(this->get_logger(), "Received: %f", msg->num);
  }

};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<MyMsgSubscriber>());
  rclcpp::shutdown();
  return 0;
}

File: src/ros_study_py/ros_study_py/my_msg_subscriber.py
import rclpy
from rclpy.node import Node
from rclpy.qos import QoSProfile
from ros_study_msgs.msg import MyMsg

class MyMsgSubscriber(Node):

    def __init__(self):
        super().__init__('my_msg_subscriber')
        qos_profile = QoSProfile(depth=10)
        self.subscription = self.create_subscription(
            MyMsg,
            'MyMsg',
            self.listener_callback,
            qos_profile
        )
        self.subscription  # prevent unused variable warning

    def listener_callback(self, msg):
        self.get_logger().info(f'Received: {msg.num}')

def main(args=None):
    rclpy.init(args=args)
    minimal_subscriber = MyMsgSubscriber()
    rclpy.spin(minimal_subscriber)
    minimal_subscriber.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
